stripe/stripe_object.py

```diff
@@ -1,17 +1,46 @@
+# pyright: strict
+from _typeshed import SupportsKeysAndGetItem
 import datetime
 import json
 from copy import deepcopy
-from typing_extensions import TYPE_CHECKING, Literal
-from typing import Any, Dict, Optional, Mapping
+from typing_extensions import TYPE_CHECKING, Literal, Self
+from typing import (
+    Any,
+    Dict,
+    List,
+    Optional,
+    Mapping,
+    Set,
+    Union,
+    cast,
+    overload,
+)
 
 import stripe
 from stripe import api_requestor, util
 
-from stripe.stripe_response import StripeResponse
+from stripe.stripe_response import StripeResponse, StripeStreamResponse
 
 
-def _compute_diff(current, previous):
+@overload
+def _compute_diff(
+    current: Dict[str, Any], previous: Optional[Dict[str, Any]]
+) -> Dict[str, Any]:
+    ...
+
+
+@overload
+def _compute_diff(
+    current: object, previous: Optional[Dict[str, Any]]
+) -> object:
+    ...
+
+
+def _compute_diff(
+    current: object, previous: Optional[Dict[str, Any]]
```

Comment:
You can make `_compute_diff` generic  based on type of `current`

Reply:
Don't know if it helps.

Reply:
Yeah I went down that road a little bit, but hit my timeout. This is fairly internal so the `Any` shouldn't leak to users.